CLAUDE.md

```
@@ -0,0 +1,1246 @@
+
+- Builder and module ordering:
+    - `core/src/test/java/tech/guilhermekaua/spigotboot/core/SpigotBootBuilderTest.java`
+- Module discovery:
+    - `core/src/test/java/tech/guilhermekaua/spigotboot/core/module/ModuleDiscoveryTest.java`
+- Package root minimization:
+    - `core/src/test/java/tech/guilhermekaua/spigotboot/core/context/lifecycle/ContextLifecyclePackageScanTest.java`
+- Configuration processing and `@Bean` proxy semantics:
+    - `core/src/test/java/tech/guilhermekaua/spigotboot/core/test/context/configuration/ConfigurationProcessorTest.java`
+- Dependency resolution and reload:
+    - `core/src/test/java/tech/guilhermekaua/spigotboot/core/test/context/dependency/manager/DependencyManagerTest.java`
+- Collection injection:
```

markdownlint-cli2 (0.21.0)

[warning] Heading style
Expected: atx; Actual: setext

(MD003, heading-style)